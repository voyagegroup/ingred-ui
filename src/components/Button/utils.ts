import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
