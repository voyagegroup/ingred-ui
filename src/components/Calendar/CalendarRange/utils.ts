import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
