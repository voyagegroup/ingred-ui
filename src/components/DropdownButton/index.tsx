export { default } from "./DropdownButton";
export type { DropdownButtonProps } from "./DropdownButton";
