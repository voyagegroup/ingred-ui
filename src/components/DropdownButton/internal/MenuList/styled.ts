import styled from "styled-components";

export const Container = styled.div<{ inline: boolean }>`
  display: ${({ inline }) => (inline ? "inline-block" : "block")};
  padding: 10px 0;
  box-shadow: 0px 0px 16px #959fa967;
  border-radius: 4px;
  background-color: #fff;
`;

export const Div = styled.div`
  cursor: pointer;
  height: 32px;
  padding: 0 16px;
  line-height: 32px;
  &:hover {
    background: #eff1f2 0% 0% no-repeat padding-box;
  }
  &:active {
    background-color: ${({ theme }) => theme.palette.primary.main};
  }
`;
