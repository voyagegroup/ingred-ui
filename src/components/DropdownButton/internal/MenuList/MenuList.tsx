import React from "react";
import * as Styled from "./styled";
import { Divider } from "./internal/Divider";

export type ContentProp = {
  text: string;
  onClick: () => void;
  divideBottom?: boolean;
};

type Props = {
  inline?: boolean;
  contents: ContentProp[];
};

const MenuList: React.FC<Props> = ({
  inline = false,
  contents,
}) => (
  <Styled.Container inline={inline}>
    {contents.map((content) => (
      <React.Fragment key={content.text}>
        <Styled.Div onClick={content.onClick}>
          {content.text}
        </Styled.Div>
        { content.divideBottom && <Divider /> }
      </React.Fragment>
    ))}
  </Styled.Container>
);

export { MenuList };
