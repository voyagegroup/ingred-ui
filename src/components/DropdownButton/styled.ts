import styled from "styled-components";
import Button from "../Button";

export const ButtonContainer = styled.div`
  display: inline-flex;
`;

export const MainButton = styled(Button)`
  border-top-right-radius: 0px;
  border-bottom-right-radius: 0px;
  border-right: none;
`;

export const SplitToggle = styled(Button)`
  border-top-left-radius: 0px;
  border-bottom-left-radius: 0px;
  border-left: 1px solid
    ${({ theme, color }) =>
      color === "primary"
        ? theme.palette.primary.deepDark
        : theme.palette.divider};
  padding: 0 ${({ theme, size }) => (size === "small" ? 0 : theme.spacing)}px;
  min-width: auto;

  &:active:not([disabled]) {
    padding-bottom: 0;
    padding-top: 2px;
  }

  &:disabled {
    border-left: 1px solid ${({ theme }) => theme.palette.divider};
  }
`;

export const SingleButton = styled(Button)`
  padding-right: ${({ theme, size }) =>
    size === "small" ? theme.spacing / 2 : theme.spacing}px;
  min-width: auto;
`;
