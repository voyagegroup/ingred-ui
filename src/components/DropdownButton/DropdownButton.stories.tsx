import * as React from "react";
import DropdownButton, { DropdownButtonProps } from "./";
import { Story } from "@storybook/react/types-6-0";
import { Flex, Spacer, Typography } from "..";
import { ContentProp } from "../MenuList/MenuList";

export default {
  title: "Components/Inputs/DropdownButton",
  component: DropdownButton,
};

const contents: ContentProp[] = [
  {
    text: "Save",
    onClick: () => {},
    type: "default",
  },
  {
    text: "Save and execute",
    onClick: () => {},
    divideTop: true,
    type: "default",
  },
  {
    text: "Save as draft",
    onClick: () => {},
    type: "default",
  },
  {
    text: "Delete",
    onClick: () => {},
    divideTop: true,
    type: "default",
  },
];

export const Example: Story<DropdownButtonProps> = (args) => (
  <DropdownButton {...args}>Click me!!</DropdownButton>
);

Example.args = {
  contents,
};

export const Colors: Story<DropdownButtonProps> = () => (
  <Flex display="flex">
    <div>
      <Typography weight="bold">Primary</Typography>
      <Spacer pt={2} />
      <DropdownButton color="primary" contents={contents}>
        Primary
      </DropdownButton>
    </div>
    <Spacer pl={3} />
    <div>
      <Typography weight="bold">Secondary</Typography>
      <Spacer pt={2} />
      <DropdownButton color="secondary" contents={contents}>
        Secondary
      </DropdownButton>
    </div>
  </Flex>
);

export const Sizes: Story<DropdownButtonProps> = () => (
  <Flex display="flex">
    <div>
      <Typography weight="bold">Small</Typography>
      <Spacer pt={2} />
      <DropdownButton size="small" contents={contents}>
        Small
      </DropdownButton>
    </div>
    <Spacer pl={3} />
    <div>
      <Typography weight="bold">Medium</Typography>
      <Spacer pt={2} />
      <DropdownButton size="medium" contents={contents}>
        Medium
      </DropdownButton>
    </div>
    <Spacer pl={3} />
    <div>
      <Typography weight="bold">Large</Typography>
      <Spacer pt={2} />
      <DropdownButton size="large" contents={contents}>
        Large
      </DropdownButton>
    </div>
  </Flex>
);

export const Split: Story<DropdownButtonProps> = () => (
  <Flex display="flex">
    <div>
      <Typography weight="bold">Normal</Typography>
      <Spacer pt={2} />
      <DropdownButton contents={contents}>Normal</DropdownButton>
    </div>
    <Spacer pl={3} />
    <div>
      <Typography weight="bold">Split</Typography>
      <Spacer pt={2} />
      <DropdownButton split={true} contents={contents}>
        Split
      </DropdownButton>
    </div>
  </Flex>
);
