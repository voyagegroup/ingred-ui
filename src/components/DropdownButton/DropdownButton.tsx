import * as React from "react";
import * as Styled from "./styled";
import { useTheme } from "../../themes";
import Icon from "../Icon";
import Spacer from "../Spacer";
import Menu, { MenuProps } from "../Menu";
import { ButtonSize, ButtonColor } from "../Button/Button";
import { ContentProp } from "../MenuList/MenuList";
import { createChainedFunction } from "../../utils/createChainedFunction";
import { useMergeRefs } from "../../hooks/useMergeRefs";
import { Placement } from "@floating-ui/react";
import { AutoPlacement } from "../../hooks/usePlacement";

type DropdownButtonColor = Exclude<ButtonColor, "danger">;

export type DropdownButtonProps = {
  size?: ButtonSize;
  color?: DropdownButtonColor;
  onClick?: () => void;
  split?: boolean;
  /**
   * props of [MenuList](/?path=/docs/components-navigation-menulist)
   */
  contents: ContentProp[];
  disabled?: boolean;
  /**
   * Define priority of position. Please check [this](https://floating-ui.com/docs/tutorial#placements).
   * For backward compatibility, `"auto" | "auto-start" | "auto-end"` are included in addition to the above positions.
   */
  positionPriority?: (Placement | AutoPlacement)[];
  menuMaxHeight?: MenuProps["maxHeight"];
  menuProps?: Partial<MenuProps>;
  children?: React.ReactNode;
};

const DropdownButton = React.forwardRef<HTMLDivElement, DropdownButtonProps>(
  function DropdownButton(
    {
      size = "medium",
      color = "primary",
      onClick,
      split = false,
      contents,
      disabled = false,
      positionPriority = ["bottom-start", "bottom-end", "top-start", "top-end"],
      menuMaxHeight = "none",
      menuProps,
      children,
    },
    ref,
  ) {
    const theme = useTheme();
    const [buttonElement, setButtonElement] =
      React.useState<HTMLDivElement | null>(null);
    const [showContent, setShowContent] = React.useState<boolean>(false);

    const handleToggleContent = (showContent: boolean) => () => {
      if (showContent && !split && onClick) {
        onClick();
      }
      setShowContent(showContent);
    };

    const setIconColor = (disabled: boolean, color: DropdownButtonColor) => {
      if (disabled) {
        return theme.palette.text.disabled;
      } else if (color === "primary") {
        return theme.palette.white;
      } else {
        return theme.palette.black;
      }
    };

    const refs = useMergeRefs<HTMLDivElement>(ref, setButtonElement);

    return (
      <>
        <Styled.ButtonContainer ref={refs} role="button">
          {split ? (
            <>
              <Styled.MainButton
                size={size}
                color={color}
                inline={true}
                disabled={disabled}
                onClick={onClick}
              >
                {children}
              </Styled.MainButton>
              <Styled.SplitToggle
                size={size}
                color={color}
                inline={true}
                disabled={disabled}
                data-testid="menu-toggle"
                onClick={handleToggleContent(!showContent)}
              >
                <Icon
                  name={"arrow_bottom"}
                  size="md"
                  // eslint-disable-next-line react/jsx-handler-names
                  color={setIconColor(disabled, color)}
                />
              </Styled.SplitToggle>
            </>
          ) : (
            <Styled.SingleButton
              size={size}
              color={color}
              disabled={disabled}
              data-testid="menu-toggle"
              onClick={handleToggleContent(!showContent)}
            >
              {children}
              <Spacer pr={0.5} />
              <Icon
                name={"arrow_bottom"}
                size="md"
                // eslint-disable-next-line react/jsx-handler-names
                color={setIconColor(disabled, color)}
              />
            </Styled.SingleButton>
          )}
        </Styled.ButtonContainer>
        <Menu
          isOpen={showContent}
          baseElement={buttonElement}
          contents={contents}
          positionPriority={positionPriority}
          maxHeight={menuMaxHeight}
          {...menuProps}
          onClose={createChainedFunction(
            handleToggleContent(false),
            menuProps?.onClose,
          )}
        />
      </>
    );
  },
);

export default DropdownButton;
