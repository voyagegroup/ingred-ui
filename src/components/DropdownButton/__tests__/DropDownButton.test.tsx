import * as React from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, act, fireEvent } from "@testing-library/react";
import DropdownButton from "..";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";
import { ContentProp } from "../../MenuList/MenuList";

jest.mock("react-dom", () => {
  const original = jest.requireActual("react-dom");
  return {
    ...original,
    createPortal: (node: any) => node,
  };
});

const contents: ContentProp[] = [
  {
    text: "Save",
    onClick: () => {},
    divideTop: true,
    type: "default",
  },
  {
    text: "Save and execute",
    onClick: () => {},
    type: "default",
  },
  {
    text: "Save as draft",
    onClick: () => {},
    divideTop: true,
    type: "default",
  },
  {
    text: "Cancel",
    onClick: () => {},
    type: "default",
  },
];

describe("DropdownButton component testing", () => {
  afterEach(cleanup);

  describe("not splited", () => {
    test("enable", async () => {
      const { asFragment, getByTestId } = renderWithThemeProvider(
        <DropdownButton contents={contents}>Save</DropdownButton>,
      );
      await act(async () => {
        fireEvent.click(getByTestId("menu-toggle"));
      });
      expect(asFragment()).toMatchSnapshot();
    });

    test("disable", () => {
      const { asFragment } = renderWithThemeProvider(
        <DropdownButton disabled={true} contents={contents}>
          Save
        </DropdownButton>,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("splited", () => {
    test("enable", async () => {
      const { asFragment, getByTestId } = renderWithThemeProvider(
        <DropdownButton split={true} contents={contents}>
          Save
        </DropdownButton>,
      );
      await act(async () => {
        fireEvent.click(getByTestId("menu-toggle"));
      });
      expect(asFragment()).toMatchSnapshot();
    });

    test("disable", () => {
      const { asFragment } = renderWithThemeProvider(
        <DropdownButton split={true} disabled={true} contents={contents}>
          Save
        </DropdownButton>,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
