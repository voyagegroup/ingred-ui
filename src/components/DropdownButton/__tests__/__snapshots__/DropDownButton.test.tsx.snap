// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownButton component testing not splited disable 1`] = `
<DocumentFragment>
  <div
    class="sc-hKwDye fsrRBT"
    role="button"
  >
    <button
      class="sc-gsDKAQ sc-dkPtRN iVvbbF govCnj sc-gKclnd bDhdBZ"
      data-testid="menu-toggle"
      disabled=""
      font-size="14px"
      font-weight="bold"
    >
      Save
      <div
        class="sc-furwcr cmiBqZ"
      />
      <div
        class="sc-iCfMLu bdvlpC"
        size="18"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0,0H24V24H0Z"
            fill="none"
          />
          <path
            d="M13.414,16.886,7.757,11.228,9.644,9.343l3.771,3.772,3.771-3.772,1.887,1.885Z"
            fill="#B3BAC1"
            transform="translate(-1.414 -1.114)"
          />
        </svg>
      </div>
    </button>
  </div>
  <div
    class="sc-bqiRlB fBQpoB"
  >
    <div
      class="sc-egiyK AIhFW"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="sc-pVTFL jQotYl"
      >
        <hr
          class="sc-crHmcD QBZFs"
          color="#E2E8EA"
          orientation="horizontal"
        />
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save
          </p>
        </div>
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save and execute
          </p>
        </div>
        <hr
          class="sc-crHmcD QBZFs"
          color="#E2E8EA"
          orientation="horizontal"
        />
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save as draft
          </p>
        </div>
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Cancel
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DropdownButton component testing not splited enable 1`] = `
<DocumentFragment>
  <div
    class="sc-hKwDye fsrRBT"
    role="button"
  >
    <button
      class="sc-gsDKAQ sc-dkPtRN iVvbbF klzIpQ sc-gKclnd bDhdBZ"
      data-testid="menu-toggle"
      font-size="14px"
      font-weight="bold"
    >
      Save
      <div
        class="sc-furwcr cmiBqZ"
      />
      <div
        class="sc-iCfMLu bdvlpC"
        size="18"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0,0H24V24H0Z"
            fill="none"
          />
          <path
            d="M13.414,16.886,7.757,11.228,9.644,9.343l3.771,3.772,3.771-3.772,1.887,1.885Z"
            fill="#FDFEFF"
            transform="translate(-1.414 -1.114)"
          />
        </svg>
      </div>
    </button>
  </div>
  <div
    class="sc-bqiRlB fgnFUF"
  >
    <div
      class="sc-hBUSln kZfwTK fade-transition-enter fade-transition-enter-active"
    >
      <div
        class="sc-ksdxgE hJTxxA"
      />
    </div>
    <div
      class="sc-egiyK AIhFW"
      data-popper-escaped="true"
      data-popper-placement="bottom-start"
      data-popper-reference-hidden="true"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="sc-pVTFL jQotYl"
      >
        <hr
          class="sc-crHmcD QBZFs"
          color="#E2E8EA"
          orientation="horizontal"
        />
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save
          </p>
        </div>
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save and execute
          </p>
        </div>
        <hr
          class="sc-crHmcD QBZFs"
          color="#E2E8EA"
          orientation="horizontal"
        />
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save as draft
          </p>
        </div>
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Cancel
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DropdownButton component testing splited disable 1`] = `
<DocumentFragment>
  <div
    class="sc-hKwDye fsrRBT"
    role="button"
  >
    <button
      class="sc-gsDKAQ sc-dkPtRN iVvbbF girpac sc-eCImPb cXxTDV"
      disabled=""
      font-size="14px"
      font-weight="bold"
    >
      Save
    </button>
    <button
      class="sc-gsDKAQ sc-dkPtRN iVvbbF girpac sc-jRQBWg chxDUC"
      data-testid="menu-toggle"
      disabled=""
      font-size="14px"
      font-weight="bold"
    >
      <div
        class="sc-iCfMLu bdvlpC"
        size="18"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0,0H24V24H0Z"
            fill="none"
          />
          <path
            d="M13.414,16.886,7.757,11.228,9.644,9.343l3.771,3.772,3.771-3.772,1.887,1.885Z"
            fill="#B3BAC1"
            transform="translate(-1.414 -1.114)"
          />
        </svg>
      </div>
    </button>
  </div>
  <div
    class="sc-bqiRlB fBQpoB"
  >
    <div
      class="sc-egiyK AIhFW"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="sc-pVTFL jQotYl"
      >
        <hr
          class="sc-crHmcD QBZFs"
          color="#E2E8EA"
          orientation="horizontal"
        />
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save
          </p>
        </div>
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save and execute
          </p>
        </div>
        <hr
          class="sc-crHmcD QBZFs"
          color="#E2E8EA"
          orientation="horizontal"
        />
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save as draft
          </p>
        </div>
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Cancel
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`DropdownButton component testing splited enable 1`] = `
<DocumentFragment>
  <div
    class="sc-hKwDye fsrRBT"
    role="button"
  >
    <button
      class="sc-gsDKAQ sc-dkPtRN iVvbbF hrIhob sc-eCImPb cXxTDV"
      font-size="14px"
      font-weight="bold"
    >
      Save
    </button>
    <button
      class="sc-gsDKAQ sc-dkPtRN iVvbbF hrIhob sc-jRQBWg chxDUC"
      data-testid="menu-toggle"
      font-size="14px"
      font-weight="bold"
    >
      <div
        class="sc-iCfMLu bdvlpC"
        size="18"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0,0H24V24H0Z"
            fill="none"
          />
          <path
            d="M13.414,16.886,7.757,11.228,9.644,9.343l3.771,3.772,3.771-3.772,1.887,1.885Z"
            fill="#FDFEFF"
            transform="translate(-1.414 -1.114)"
          />
        </svg>
      </div>
    </button>
  </div>
  <div
    class="sc-bqiRlB fgnFUF"
  >
    <div
      class="sc-hBUSln kZfwTK fade-transition-enter fade-transition-enter-active"
    >
      <div
        class="sc-ksdxgE hJTxxA"
      />
    </div>
    <div
      class="sc-egiyK AIhFW"
      data-popper-escaped="true"
      data-popper-placement="bottom-start"
      data-popper-reference-hidden="true"
      style="position: absolute; left: 0px; top: 0px;"
    >
      <div
        class="sc-pVTFL jQotYl"
      >
        <hr
          class="sc-crHmcD QBZFs"
          color="#E2E8EA"
          orientation="horizontal"
        />
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save
          </p>
        </div>
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save and execute
          </p>
        </div>
        <hr
          class="sc-crHmcD QBZFs"
          color="#E2E8EA"
          orientation="horizontal"
        />
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Save as draft
          </p>
        </div>
        <div
          class="sc-iqseJM eUya-Dg"
        >
          <p
            class="sc-bdvvtL jiNrDF sc-kDTinF"
            color="#041C33"
            font-size="13px"
          >
            Cancel
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
